src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1511,62 +1470,138 @@ private BoundMethodOrPropertyGroup ConstructBoundMemberGroupAndReportOmittedType
             }
         }
 
-        private bool IsPossiblyCapturingPrimaryConstructorParameterReference(BoundExpression colorColorValueReceiver, out ParameterSymbol parameterSymbol)
+        private BoundExpression CheckAmbiguousPrimaryConstructorParameterAsColorColorReceiver(BoundExpression receiver, string plainName, ImmutableArray<TypeWithAnnotations> typeArguments, bool invoked, ArrayBuilder<Symbol> members, BindingDiagnosticBag diagnostics)
```

Comment:
:memo: extracted the logic and moved the check out of `ConstructBoundMemberGroupAndReportOmittedTypeArguments` (the caller is responsible to call `CheckAmbiguousPrimaryConstructorParameterAsColorColorReceiver`)